tests/driver/index.test.ts

```diff
@@ -36,235 +36,289 @@ describe(`Driver based tests`, async () => {
+        const product1 = new Product({
+          name: 'Product 1',
+          price: 10,
+          expiryDate: new Date('2024-04-20T00:00:00.000Z'),
+          isCertified: true
+        });
+        await product1.save();
 
-      await cart.deleteOne();
-      assert.strictEqual(await Cart.findOne({ cartName: 'wewson' }), null);
+        const product2 = new Product({
+          name: 'Product 2',
+          price: 10,
+          expiryDate: new Date('2024-11-20T00:00:00.000Z'),
+          isCertified: false
+        });
+        await product2.save();
 
-      if (isAstra) {
-        astraMongoose?.connection.dropCollection('carts');
-        astraMongoose?.connection.dropCollection('products');
-      } else {
-        jsonAPIMongoose?.connection.dropCollection('carts');
-        jsonAPIMongoose?.connection.dropCollection('products');
+        const cart = new Cart({
+          name: 'My Cart',
+          cartName: 'wewson',
+          products: [product1._id, product2._id]
+        });
+        await cart.save();
+        assert.strictEqual(await Cart.findOne({cartName: 'wewson'}).select('name').exec().then((doc: any) => doc.name), cart.name);
+        //compare if product expiryDate is same as saved
+        assert.strictEqual(await Product.findOne({name: 'Product 1'}).select('expiryDate').exec().then((doc: any) => doc.expiryDate.toISOString()), product1.expiryDate!.toISOString());
+
+        const findOneAndReplaceResp = await Cart.findOneAndReplace({cartName: 'wewson'}, {
+          name: 'My Cart 2',
+          cartName: 'wewson1'
+        }, {returnDocument: 'after'}).exec();
+        assert.strictEqual(findOneAndReplaceResp!.name, 'My Cart 2');
+        assert.strictEqual(findOneAndReplaceResp!.cartName, 'wewson1');
+
+        let productNames: string[] = [];
+        const cursor = await Product.find().cursor();
+        await cursor.eachAsync(p => productNames.push(p.name));
+        assert.deepEqual(productNames.sort(), ['Product 1', 'Product 2']);
+
+        await cart.deleteOne();
+        assert.strictEqual(await Cart.findOne({cartName: 'wewson'}), null);
+      } finally {
+        if (isAstra) {
+          astraMongoose?.connection.dropCollection('carts');
+          astraMongoose?.connection.dropCollection('products');
+        } else {
+          jsonAPIMongoose?.connection.dropCollection('carts');
+          jsonAPIMongoose?.connection.dropCollection('products');
+        }
       }
     });
   });
   describe('Mongoose API', () => {
     it('handles find cursors', async () => {
-      const mongooseInstance = await createMongooseInstance();
-
-      const personSchema = new mongooseInstance.Schema({
-        name: String
-      });
-      const Person = mongooseInstance.model('Person', personSchema);
-      await Person.init();
-      await Person.deleteMany({});
-      await Person.create([{ name: 'John' }, { name: 'Bill' }]);
+      let mongooseInstance;
```

Comment:
Similar here. Let's create `mongooseInstance` in a `beforeEach()`, and call `await mongooseInstance?.connection.dropCollection('people');` in an `afterEach()`.

Reply:
Makes sense. I did this way since the collections used by different tests were not same. Now, I moved the connection step to beforeEach and moved the dropCollection part from the finally to afterEach (to have all unique collections that are used across tests).